### ISSUE TITLE
"failed assertion: checkModel(o)" error when joining over more than three relations

### ISSUE THREAD

**Issue by thuwuyinjun** [IITDBGroup/gprom#48]
Hello, Prof. Glavic!
     This is Yinjun who has ever contacted you in terms of code issue before. Thanks very much again for your instructions on how to use gprom. But I have to say sorry since I need to disturb you again since there is another minor issue now....
     I am currently using Postgresql database as the underlying database. This is the URL for database I used (please copy the link and paste it in the browser): [https://drive.google.com/file/d/1pDYU0SBqT81sBSu7AyYQtDrVLd3tjTTq/view?usp=sharing](url)

If you run the following query for provenance against this database:
PROVENANCE OF (select "n2"."identifier","n2"."name" from "gene" "n1","participates_gpbp_gene_biologicalprocess" "e2","biologicalprocess" "n2","participates_gpbp_gene_biologicalprocess" "e3" where "n1"."gene_id"='0' and "n2"."biologicalprocess_id"='2' and "e2"."biologicalprocess_id"='2' and "e3"."biologicalprocess_id"='2')

GPROM will throw an error, i.e. failed assertion: checkModel(o). You can find the details of this error in the following figure.
![gprom_error](https://user-images.githubusercontent.com/8242415/41737556-a5cdc500-755d-11e8-93f3-f2b46e9715b2.png)

This is a query including 4 relations. But when I remove one of any relations and corresponding predicates in the where clause, the error disappears. I tried it several times. It seems that when there are more than 3 relations, the assertion error will be triggered. Does that mean that gprom cannot support queries with too many relations?

Could you please help me figure out what happened in this example? I really appreciate your help!

Thanks!

**Comment by lordpretzel** [comment]
This is related to some automated renaming to disambiguate column names in join results which seems to fail after two joins. A minimal test case:

~~~
SELECT x1.a, x2.a AS b FROM r x1, r x2, r x3, r x4;
~~~

for 

~~~
CREATE TABLE r (a int);
~~~

**Comment by lordpretzel** [comment]
fixed in 30117af1d49fe795613098f6b737eb4e36d4446b

**Comment by thuwuyinjun** [comment]
It works. Thanks!
